presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -490,6 +490,29 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                     combineConjuncts(postJoinConjuncts.build()));
         }
 
+        private boolean shouldRemoveOuterJoinConjunct(Collection<Symbol> outerSymbols, Expression conjunct)
+        {
+
+            if (!(conjunct instanceof ComparisonExpression)) {
```

Comment:
if conjuct is not `ComparisonExpression` or at least it's not equality comparison and it touches both sides of the table I believe it should still stay? Or not?